### PR TITLE
Add `instance_type_list` module

### PR DESCRIPTION
**Pull request by ezilber-akamai**
## 📝 Description

Add `instance_type_list` module

## ✔️ How to Test

`make TEST_ARGS="-v instance_type_list" test`

### CODE REVIEW

**Comment by ezilber-akamai** [comment]
/acctest tests=instance_type_list sha=7f66a8f4df672492f55863c0c45cbaf43b8a9245

**Review by zliang-akamai — APPROVED** [review]

File: plugins/modules/instance_type_list.py [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+SPECDOC_META = SpecDocMeta(
+    description=[
+        'List and filter on Linode Instance Types.'
+    ],
+    requirements=global_requirements,
+    author=global_authors,
+    options=spec,
+    examples=docs.specdoc_examples,
+    return_values=dict(
+        instanceTypes=SpecReturnValue(
```
Comment (lgarber-akamai) [inline plugins/modules/instance_type_list.py:62]: Looks like this should be `instance_types`

File: plugins/module_utils/doc_fragments/instance_type_list.py [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+specdoc_examples = ['''
+- name: List all of the Linode instance types
+  linode.cloud.instance_type_list: {}''', '''
+- name: Resolve all Linode instance types
+  linode.cloud.instance_type_list:
+    filter:
+      - name: class
+        values: nanode''']
+
+result_images_samples = ['''[
```
Comment (lgarber-akamai) [inline plugins/module_utils/doc_fragments/instance_type_list.py:12]: Quick correction just for consistency:
`result_images_samples` -> `result_instance_types_samples`

**Review by lgarber-akamai — CHANGES_REQUESTED** [review]
Great work! Just a few small typo corrections

**Review by lgarber-akamai — APPROVED** [review]

**Pull request merged**